src/middlewares/auth.ts

```diff
@@ -0,0 +1,23 @@
+import { NextFunction, Request, Response } from 'express';
+
+import Firebase from 'src/config/firebase';
+
+export interface RequestWithFirebase extends Request {
+  firebaseUid?: string;
+}
+
+export const checkToken = async (req: RequestWithFirebase, res: Response, next: NextFunction) => {
+  if (!req.headers.authorization) {
+    return res.boom.badRequest('Token is required');
+  }
+
+  return Firebase.auth()
```

Comment:
using async...await might be better for maintainability and scalability issues

Reply:
Done.